### PR TITLE
Added enformer predictor

### PR DESCRIPTION
**Pull request by gtsitsiridis**

### CODE REVIEW

File: envs/abexp-enformer-gpu.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  - polars>=0.20.19
+  - scikit-learn>=1.4.1.post1
+  - lightgbm>=4.3.0
+  - zarr>=2.15.0
+  - xarray>=2024.5.0
+  - tensorflow-hub==0.16.1
+  - tensorflow==2.16.1
+  - tensorflow-gpu==2.16.1
+  - pip:
+    - "git+https://github.com/gtsitsiridis/kipoi_veff_analysis.git@master"
```
Comment (Hoeze) [inline envs/abexp-enformer-gpu.yaml]: We need to move this repo to the gagneurlab team

File: envs/abexp-enformer.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  - pyranges>=0.0.129
+  - polars>=0.20.19
+  - scikit-learn>=1.4.1.post1
+  - lightgbm>=4.3.0
+  - zarr>=2.15.0
+  - xarray>=2024.5.0
+  - tensorflow-hub==0.16.1
+  - tensorflow==2.16.1
+  - pip:
+    - "git+https://github.com/gtsitsiridis/kipoi_veff_analysis.git@master"
```
Comment (Hoeze) [inline envs/abexp-enformer.yaml]: move to gagneurlab

File: scripts/fset@abexp_v1.0.yaml
Comment (Hoeze) [inline scripts/fset@abexp_v1.0.yaml:1]: please revert changes to this file. AbExp v1.0 cannot handle Enformer

File: scripts/veff/enformer/scripts/gtf2parquet.py
Comment (Hoeze) [inline scripts/veff/enformer/scripts/gtf2parquet.py:1]: Please check if this rule can be replaced with this one:
https://github.com/gagneurlab/AbExp/tree/main/scripts/gtf_transcripts.py.py

File: scripts/veff/enformer/enformer_vcf.smk [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    output:
+        temp(RAW_VCF_PQ_PATTERN)
+    input:
+        gtf_path=rules.enformer_gtf_to_parquet.output[0],
+        fasta_path=FASTA_FILE,
+        vcf_path=STRIPPED_VCF_FILE_PATTERN
+    params:
+        type='alternative'
+    conda:
+        # requires GPU on environment installation
```
Comment (Hoeze) [inline scripts/veff/enformer/enformer_vcf.smk]: why does it require a GPU on installation?

**Review by Hoeze — CHANGES_REQUESTED** [review]
Looks good! Overall quite clean.
I made some comments and asked for minor changes.
We also need to put your veff repo into the Gagneurlab org, and maybe change its name...

File: scripts/fset@abexp_v1.0.yaml
Comment (gtsitsiridis) [reply scripts/fset@abexp_v1.0.yaml:1]: Should I create a new file with a different version?

Comment (Hoeze) [reply scripts/fset@abexp_v1.0.yaml:1]: No, I already have one on my side. I need to test your code anyways :)

File: scripts/veff/enformer/enformer_vcf.smk (hunk 3 above)
Comment (gtsitsiridis) [reply scripts/veff/enformer/enformer_vcf.smk]: I forgot that erase this comment. We only need a GPU if we want to install tensorflow with GPU support (i.e. use the GPU conda environment).

File: scripts/veff/enformer/scripts/gtf2parquet.py
Comment (gtsitsiridis) [reply scripts/veff/enformer/scripts/gtf2parquet.py:1]: yes it can

**Comment by gtsitsiridis** [comment]
I addressed the comments above. We still need to move the repo to the gagneurlab team.

**Pull request merged**